src/hv_anndata/components.py

```diff
@@ -0,0 +1,144 @@
+            self.param._input_value,
+            options=self.param.options,
+            placeholder="Enter/select value",
+            name="Available values",
+            restrict=False,
+            min_characters=0,
+            disabled=self.param._input_key.rx().rx.bool().rx.not_(),
+            sizing_mode="stretch_width",
+        )
+
+        self._multi_choice = pmui.MultiChoice.from_param(
+            self.param._current_selection,
+            options=self.param.options,
+            name="Values for the selected group",
+            searchable=True,
+            disabled=self._value_input.param.disabled,
+            sizing_mode="stretch_width",
+        )
+
+        self._json_editor = pn.widgets.JSONEditor.from_param(
+            self.param.value,
+            name="JSON Editor",
+            mode="tree",
+            menu=False,
+            sizing_mode="stretch_width",
+        )
+
+    @param.depends("_input_key", watch=True)
+    def _handle_key_input(self) -> None:
+        """Handle when a key is entered in the text input."""
+        key = self._input_key.strip()
+        if not key:
+            return
+        # Set the current key
+        self._current_key = key
+
+        # Initialize the key in the value dict if it doesn't exist
+        if key not in self.value:
+            new_value = dict(self.value)
+            new_value[key] = []
+            self.value = new_value
+
+        # Update current selection to match the key's current values
+        self._current_selection = list(self.value.get(key, []))
+
+        if key not in self._key_input.options:
+            self._key_input.options = [*self._key_input.options, key]
+
+    @param.depends("_input_value", watch=True)
+    def _handle_value_input(self) -> None:
+        """Handle when a value is entered in the text input."""
+        value = self._input_value.strip()
+        if not value or not self._current_key:
+            return
+
+        if value not in self.options:
+            self.options = [*self.options, value]
+
+        # Add the value to the current selection for the active key
+        if value not in self._current_selection:
+            self._current_selection = [*self._current_selection, value]
+
+        self._value_input.value = ""
+
+    @param.depends("_current_selection", watch=True)
+    def _handle_selection_change(self) -> None:
+        """Handle when the MultiChoice selection changes."""
+        if not self._current_key:
+            return
+
+        # Update the value dict with the new selection
+        new_value = dict(self.value)
+        new_value[self._current_key] = list(self._current_selection)
+        self.value = new_value
+
+    def __panel__(self) -> pn.layout.Column:
+        return pn.Column(
+            self._key_input,
+            self._value_input,
+            self._multi_choice,
+            self._json_editor,
+        )
+
+
+if pn.state.served:
```

Comment:
Woah, I've never seen this pattern before. Is this similar to `if __name__ == "__main__"`?

I couldn't find any docs on it https://panel.holoviz.org/search.html?q=pn.state.served

Reply:
Yes, at least that's the way I see it. `__name__ == "__main__"` doesn't evaluate to True when running a script with `panel serve ...` but `pn.state.served` does. It has some gotchas https://github.com/holoviz/panel/issues/5623.